builder/azure/arm/config.go

```diff
@@ -129,11 +130,12 @@ type Config struct {
 	TempResourceGroupName             string             `mapstructure:"temp_resource_group_name"`
 	BuildResourceGroupName            string             `mapstructure:"build_resource_group_name"`
 	storageAccountBlobEndpoint        string
-	PrivateVirtualNetworkWithPublicIp bool   `mapstructure:"private_virtual_network_with_public_ip"`
-	VirtualNetworkName                string `mapstructure:"virtual_network_name"`
-	VirtualNetworkSubnetName          string `mapstructure:"virtual_network_subnet_name"`
-	VirtualNetworkResourceGroupName   string `mapstructure:"virtual_network_resource_group_name"`
-	CustomDataFile                    string `mapstructure:"custom_data_file"`
+	PrivateVirtualNetworkWithPublicIp bool     `mapstructure:"private_virtual_network_with_public_ip"`
+	VirtualNetworkName                string   `mapstructure:"virtual_network_name"`
+	VirtualNetworkSubnetName          string   `mapstructure:"virtual_network_subnet_name"`
+	VirtualNetworkResourceGroupName   string   `mapstructure:"virtual_network_resource_group_name"`
+	AllowedInboundIpAddresses         []string `mapstructure:"allowed_inbound_ip_addresses"`
```

Comment:
>AllowedInboundIpAddresses [](start = 1, length = 25)

Are you planning to allow CIDRs? or just individual IPs (/32)?

Reply:
Well, I was thinking IP addresses only but allowing CIDRs are straight forward as well - both from config point of view (this PR) and an actual ARM template point of view.
So I'll update this PR to support CIDRs as well.

Reply:
Left the property name unchanged (allowed_inbound_ip_addresses) rather than changing it to something more verbose - like allowed_inbound_ip_addresses_cidrs
This is because, IMO, it's still a list of IP addresses.
Please let me know if you disagree.

Reply:
Looks great!